### PR TITLE
Simplify POC installation

### PR DESCRIPTION
**Pull request by liu-cong**
This was adapted from the private POC repo.

This PR separates what needs to installed vs. what resources user creates as sample applications. The installation is simplified to a single gatewayclass.yaml file.

@courageJ encountered some issue when trying the POC, and this PR worked for her. Not sure what didn't work before though :)

### CODE REVIEW

File: examples/poc/manifests/gatewayclass.yaml [hunk 1]
```diff
@@ -88,28 +90,66 @@ spec:
...
-    name: custom-proxy-config
-    namespace: envoy-gateway-system
+    name: llm-route-envoy-config
+    namespace: llm-gateway
+
 ---
-apiVersion: gateway.networking.k8s.io/v1
-kind: Gateway
+apiVersion: apps/v1
+kind: Deployment
```
Comment (kfswain) [inline examples/poc/manifests/gatewayclass.yaml:110]: gatewayclass as the file name may be misleading if we are putting the ext_proc deployment yaml here

**Review by kfswain — COMMENTED** [review]
LGTM with the assumption that this has been validated E2E

File: examples/poc/manifests/gatewayclass.yaml (hunk 1 above)
Comment (liu-cong) [reply examples/poc/manifests/gatewayclass.yaml:110]: The point is that ext_proc is an implementation detail of this specific gateway class

**Comment by liu-cong** [comment]
> LGTM with the assumption that this has been validated E2E

Yes both @courageJ and I tried this.

File: examples/poc/manifests/gatewayclass.yaml (hunk 1 above)
Comment (kfswain) [reply examples/poc/manifests/gatewayclass.yaml:110]: A reasonable perspective. 

However, the core of this PoC _is_ the ext-proc implementation, and I don't think can be considered an implementation detail here.

Comment (liu-cong) [reply examples/poc/manifests/gatewayclass.yaml:110]: how about just call it installation.yaml which just bundles everything, or can break it down to multiple yaml files

Comment (kfswain) [reply examples/poc/manifests/gatewayclass.yaml:110]: Either SGTM, thanks!

Comment (liu-cong) [reply examples/poc/manifests/gatewayclass.yaml]: cool, renamed to installation.yaml. And the README explains what's being installed so this should be good!

File: examples/poc/README.md [hunk 2]
```diff
@@ -1,68 +1,55 @@
...
-
-- The Go application collects metrics and saves the latest response headers in memory.
-- Ensure Envoy is configured to route based on the metrics collected from the `/metric` endpoint of different service pods.
-
-## Contributing
+4. **Try it out** 
+   Wait until the gateway is ready.
+   ```bash
+   IP=$(kubectl get gateway/llm-gateway -o jsonpath='{.status.addresses[0].value}')
+   PORT=8081
```
Comment (Joffref) [inline examples/poc/README.md:42]: Isn't this wrong, llm-instance-gw is listening on 8080. Am I wrong?

**Comment by Joffref** [comment]
Overall it looks good to me and it's a great addition. Thank you @liu-cong 😀

File: examples/poc/README.md (hunk 2 above)
Comment (liu-cong) [reply examples/poc/README.md:42]: Good question! Actually in the POC setup, Envoy is configured the additional 8081 port for ext proc traffic. Updated README

Comment (Joffref) [reply examples/poc/README.md:42]: Ho yes, you're right actually. Sorry, for the confusion.

File: examples/poc/README.md [hunk 3]
```diff
@@ -1,68 +1,55 @@
 # Envoy Ext Proc Gateway with LoRA Integration
 
-This project sets up an Envoy gateway to handle gRPC calls with integration of LoRA (Low-Rank Adaptation). The configuration aims to manage gRPC traffic through Envoy's external processing and custom routing based on headers and load balancing rules. The setup includes Kubernetes services and deployments for both the gRPC server and the vllm-lora application.
+This project sets up an Envoy gateway with a custom external processing which  implements advanced routing logic tailored for LoRA (Low-Rank Adaptation) adapters. The routing algorithm is based on the model specified (using Open AI API format), and ensuring efficient load balancing based on model server metrics.
```
Comment (Xunzhuo) [inline examples/poc/README.md:3]: one more extra space between `which` and `implements`

**Review by terrytangyuan — APPROVED** [review]
/lgtm
/approve

**Pull request merged**